### PR TITLE
Aura 2.0

### LINKED ISSUE #85: -A doesn't handle metapackages
When I try to install`kdebase4-runtime-git`:

```
$ sudo aura -A kdebase4-runtime-git
aura >>= Determining dependencies...
aura >>= Dependency checking failed for these reasons:
`phonon-backend` exists in NO WAY as a package or as one provided by another!
```

However, when I try to install kdebase-runtime, pacman resolves the metapackage correctly:

```
sudo aura -S kdebase-runtime
resolving dependencies...
:: There are 2 providers available for phonon-backend:
:: Repository extra
   1) phonon-gstreamer  2) phonon-vlc

Enter a number (default=1):
```

### PR DESCRIPTION
**Pull request by fosskers**
This will be the last major Aura release that I foresee. It includes most of the improvements planned for the original Aura 1.4, including:

- [x] Use of `shelly`
- [x] CLI overhaul via `optparse-applicative`
- [x] `Text` instead of `String`
- [x] Use of the `versions` library
- [x] Use of `freer-simple` (Extensible Effects) to remove the `Aura` monad completely

As well as:

- [x] Fast, concurrent dependency resolution via `async`
- [x] A test suite for backend logic
- [x] JSON-based package states
- [x] Control over coloured output
- [x] Automatic package state saving upon `-Au`
- [x] Rework contributing guides (they're out of date)
- [x] Various bug fixes

Closes #85 #102 #158  #213 #217 #218 #296 #297 #311 #324 #346 #353 #357 #377 #400 #437 #445 #447 #451

### CODE REVIEW

File: aura/src/Aura/Core.hs [hunk 1]
```diff
@@ -21,31 +23,26 @@ along with Aura.  If not, see <http://www.gnu.org/licenses/>.
...
 
 ---
 
 --------
 -- TYPES
 --------
-type Error    = String
-type Pkgbuild = String
+
+type Pkgbuild = T.Text
```
Comment (fosskers) [inline aura/src/Aura/Core.hs]: This should be a `newtype`.

File: aura/src/Aura/Pacman.hs [hunk 2]
```diff
@@ -24,53 +25,70 @@ along with Aura.  If not, see <http://www.gnu.org/licenses/>.
...
 
-type ShellArg = String
-
-defaultCmd :: String
+defaultCmd :: T.Text
 defaultCmd = "pacman"
 
-powerPillCmd :: String
+powerPillCmd :: T.Text
 powerPillCmd = "/usr/bin/powerpill"
```
Comment (fosskers) [inline aura/src/Aura/Pacman.hs]: Is `powerpill` still around? Do we need to support it?

**Comment by fosskers** [comment]
Just finished the port to `optparse-applicative`. That was a lot of typing.

**Comment by fosskers** [comment]
Port to `freer-simple` went well. Function signatures are more honest now (albeit a bit longer).

**Review by patriciamoreirag3 — COMMENTED** [review]
Patricia Moreira Gonçalves Moura
patriciamoreirag@gmail.com

**Pull request merged**